src/gatling/simulations/scenarios/NFD_02DivorceApplication.scala

```diff
@@ -200,13 +200,21 @@ object NFD_02DivorceApplication {
     .group("DivorceApp_270_DocumentUpload") {
       exec(http("Upload your documents")
         .post(BaseURL + "/document-manager?_csrf=${csrf}")
-        .headers(CommonHeader)
-        .headers(PostHeader)
+        .header("accept", "application/json")
+        .header("accept-encoding", "gzip, deflate, br")
+        .header("accept-language", "en-GB,en;q=0.9")
+        .header("content-type", "multipart/form-data")
+        .header("sec-fetch-dest", "empty")
+        .header("sec-fetch-mode", "cors")
+        .header("sec-fetch-site", "same-origin")
+        .header("user-agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36")
+        .header("x-requested-with", "XMLHttpRequest")
         .bodyPart(RawFileBodyPart("files[]", "2MB.pdf")
           .fileName("2MB.pdf")
           .transferEncoding("binary"))
         .asMultipartForm
-        .check(status.is(200)))
+        .check(status.is(200))
+        .check(regex(""""id":"(.+)","name":"2MB.pdf"""").saveAs("documentId")))
```

Comment:
will this save the json response?

Reply:
It saves the returned Document ID, which is required for the subsequent POST.